rocketpy/simulation/dispersion.py

```diff
@@ -153,9 +156,260 @@ def __init__(
+        except (TypeError, ValueError, KeyError, AttributeError) as error:
+            print(f"Error on iteration {self.iteration_count}: {error}")
+            error_file.write(f"{self._inputs_dict}\n")
+            self.__close_files(input_file, output_file, error_file)
+            raise error
+        except KeyboardInterrupt:
+            print("Keyboard Interrupt, files saved.")
+            error_file.write(f"{self._inputs_dict}\n")
+            self.__close_files(input_file, output_file, error_file)
+
+        self.__finalize_simulation(input_file, output_file, error_file)
+
+    def __run_single_simulation(self, input_file, output_file):
+        """Runs a single simulation and saves the inputs and outputs to the
+        respective files."""
+        # Update iteration count
+        self.iteration_count += 1
+        # Run trajectory simulation
+        dispersion_flight = Flight(
+            rocket=self.rocket.create_object(),
+            environment=self.environment.create_object(),
+            rail_length=self.flight._randomize_rail_length(),
+            inclination=self.flight._randomize_inclination(),
+            heading=self.flight._randomize_heading(),
+            initial_solution=self.flight.initial_solution,
+            terminate_on_apogee=self.flight.terminate_on_apogee,
+        )
+
+        self._inputs_dict = dict(
+            item
+            for d in [
+                self.environment.last_rnd_dict,
+                self.rocket.last_rnd_dict,
+                self.flight.last_rnd_dict,
+            ]
+            for item in d.items()
+        )
+
+        # Export inputs and outputs to file
+        self.__export_flight_data(
+            flight=dispersion_flight,
+            inputs_dict=self._inputs_dict,
+            input_file=input_file,
+            output_file=output_file,
+        )
+
+        self.__reprint(
+            f"Current iteration: {self.iteration_count:06d} | "
+            f"Average Time per Iteration: {((process_time() - self.start_cpu_time) / self.iteration_count):2.6f} s | "
+            f"Estimated time left: {int((self.number_of_simulations - self.iteration_count) * ((process_time() - self.start_cpu_time) / self.iteration_count))} s",
+            end="\r",
+            flush=True,
+        )
+
+    def __close_files(self, input_file, output_file, error_file):
+        """Closes all the files."""
+        input_file.close()
+        output_file.close()
+        error_file.close()
+
+    def __finalize_simulation(self, input_file, output_file, error_file):
+        """Finalizes the simulation, closes the files and prints the results."""
+        final_string = (
+            f"Completed {self.iteration_count} iterations. Total CPU time: "
+            f"{process_time() - self.start_cpu_time:.1f} s. Total wall time: "
+            f"{time() - self.start_time:.1f} s\n"
+        )
+
+        self.__reprint(final_string + f"Saving results.", flush=True)
```

Comment:
You are using a f-string that doesn't have an interpolated value. Make it a simple string or add some content to interpolate. Here is a simple suggestion:

```suggestion
        self.__reprint(final_string + "Saving results.", flush=True)
```